### PR TITLE
[#4093] Migrate Microsoft.Bot.Builder.Dialogs.Adaptive.Tests to xUnit

### LINKED ISSUE #4093: migrate builder.dialogs.adaptive to xunit
#4039

### PR DESCRIPTION
**Pull request by denscollo**
Fixes #4093

## Description
This PR migrates all [Microsoft.Bot.Builder.Dialogs.Adaptive](https://github.com/microsoft/botbuilder-dotnet/tree/main/tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests) tests from **MSTest** to **xUnit**.

## Specific Changes
- Replaced `MSTest` packages with `xUnit` in the `.csproj`.
- Changed `[TestClass]` to `[CollectionDefinition]`.
- Changed `[TestMethod]` to `[Fact]`.
- Changed `Assert.AreEqual` to `Assert.Equal`.
- Changed `Assert.IsNotNull` to `Assert.NotNull`.
- Changed `Assert.IsNull` to `Assert.Null`.
- Added `[assembly: TestFramework("Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.Tests.Startup", "Microsoft.Bot.Builder.Dialogs.Adaptive.Tests")]` to the `Startup` class to replace `[AssemblyInitialize]`.
- Implemented `ResourceExplorerFixture` class to share the `ResourceExplorer` among tests.
- Implemented `ConfigurationFixture` class to share the `Configuration` among tests.
- Improved tests Assertions.

## Testing
In the following image there are the passing tests.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ConfigurationFixture.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        public ConfigurationFixture()
+        {
+            Configuration = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: false)
+                .Build();
+        }
+
+        public IConfiguration Configuration { get; private set; }
+
+        public void Dispose()
```
Comment (EricDahlvang) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ConfigurationFixture.cs:20]: I don't think this dispose is actually doing anything.  Is this class needed?

**Review by EricDahlvang — CHANGES_REQUESTED** [review]
This is great!  Thanks for cleaning up these tests.

I'm curious about ConfigurationFixture, since it is not necessarily adding anything.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/GeneratorTests.cs [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
 
 namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Tests
 {
-    [TestClass]
-    public class GeneratorTests
+    [CollectionDefinition("Dialogs.Adaptive")]
+    public class GeneratorTests : IClassFixture<ResourceExplorerFixture>
     {
         private readonly string sandwichDirectory = PathUtils.NormalizePath(@"..\..\..\..\..\tests\Microsoft.Bot.Builder.Dialogs.Adaptive.Tests\Tests\GeneratorTests\sandwich\");
         private readonly string unitTestsDirectory = PathUtils.NormalizePath(@"..\..\..\..\..\tests\Microsoft.Bot.Builder.Dialogs.Adaptive.Tests\Tests\GeneratorTests\unittests\");
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/GeneratorTests.cs]: Done. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/SettingsStateTests.cs [hunk 3]
```diff
@@ -1,44 +1,33 @@
...
 namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Tests
 {
-    [TestClass]
-    public class SettingsStateTests
+    [CollectionDefinition("Dialogs.Adaptive")]
+    public class SettingsStateTests : IClassFixture<ResourceExplorerFixture>, IClassFixture<ConfigurationFixture>
     {
-        public static ResourceExplorer ResourceExplorer { get; set; }
+        private readonly ResourceExplorerFixture _resourceExplorerFixture;
+        private readonly ConfigurationFixture _configurationFixture;
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/SettingsStateTests.cs]: We rollback that change leaving the IConfiguration as it was. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Startup.cs [hunk 4]
```diff
@@ -37,11 +34,18 @@ public static void Initialize(TestContext testContext)
...
+
         public void ConfigureServices(IServiceCollection services)
         {
             // Adding IConfiguration in sample test server.  Otherwise this appears to be 
             // registered.
-            services.AddSingleton<IConfiguration>(this.Configuration);
+            services.AddSingleton(Configuration);
+        }
+
+        public new void Dispose()
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Startup.cs]: Done, thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ConfigurationFixture.cs (hunk 1 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ConfigurationFixture.cs]: We removed this class. Thanks!

**Review by EricDahlvang — APPROVED** [review]

**Pull request merged**